src/output.cpp

```diff
@@ -1081,6 +1294,137 @@ void JsonOutput::map_hist(
   out_ << "}}" << std::endl;
 }
 
+void JsonOutput::map_tseries(
+    BPFtrace &bpftrace,
+    const BpfMap &map,
+    const std::map<std::vector<uint8_t>,
+                   std::map<uint64_t, std::vector<uint8_t>>> &values_by_key,
+    uint64_t latest_epoch) const
+{
+  const auto &map_info = bpftrace.resources.maps_info.at(map.name());
+
+  out_ << R"({"type": ")" << MessageType::tseries << R"(", "data": {)";
+  out_ << "\"" << json_escape(map.name()) << "\": ";
+  if (!map_info.is_scalar) // check if this map has keys
+    out_ << "{";
+
+  map_tseries_contents(bpftrace, map, values_by_key, latest_epoch);
+
+  if (!map_info.is_scalar) // check if this map has keys
+    out_ << "}";
+  out_ << "}}" << std::endl;
+}
+
+template <typename T>
+std::string tseries_to_str_json(
+    BPFtrace &bpftrace,
+    const std::map<uint64_t, std::vector<uint8_t>> &values,
+    uint64_t latest_epoch,
+    uint64_t interval_ns,
+    uint64_t buckets,
+    const SizedType &inner_type)
+{
+  std::ostringstream res;
+  bool mm_set = false;
+  uint64_t min_epoch = 0;
+  T min_value = 0;
+  T max_value = 0;
+
+  for (const auto &v : values) {
```

Comment:
Consider factoring out this logic that finds `min_epoch` and `min_value`/`max_value`. It makes up almost half of the function body of both text and json output functions.